### DISCUSSION TITLE
Test on Mac OSX 10.9

### DISCUSSION THREAD

**Discussion by sincebelieve** [bwoods/TeX--#8] [Q&A] [answered]
Hi there, I just test the project on Mac OSX 10.9, it works very well.

The only problem is that `tex.pool` is not found under `tex/`, maybe it should be moved from `web/` to `tex/`, otherwise the program utters:

```
$ ./a.out test.tex 
! I can't read tex.pool.
```

I want to port this to iOS and Android so that I can compile TeX scripts to get some images of formula. So I need to convert the `.dvi` files to `.png` or `.svg`, I'm currently investigating [dvisvgm](http://dvisvgm.sourceforge.net/) and [dvipng](http://savannah.nongnu.org/projects/dvipng/), do you know any other better options?

Thanks.

**Comment by bwoods** [comment, marked as answer]
On iOS at least, I have these Copy build steps; note the second one:

![copy files steps](https://cloud.githubusercontent.com/assets/26312/2886629/7fb07cc2-d4ea-11e3-8878-25eeb10fc9ee.png)
Perhaps I should mention this in the README.

In the source tree, `tex.pool` it is in the `web` folder because it is one of the source files manipulated by the `tex.sh` script.

Of course, in your app you can place the files wherever you need. You don’t even need to use the ones I supplied. In fact, if you plan on using more than `plainTeX` you _will_ need more than I provide.

**Comment by bwoods** [comment]
> I'm currently investigating [dvisvgm](http://dvisvgm.sourceforge.net/) and [dvipng](http://savannah.nongnu.org/projects/dvipng/), do you know any other better options?

I am eventually planning on writing a `.dvi` parser that produces `.pdf` files. But that may be a while…

Be aware that the projects you mention are GPL’d and LGPL’d respectively. `dvisvgm`, being under the GPL, [cannot be used in an App Store app](http://stackoverflow.com/a/8694538). `dvipng`, being under the LGPL, requires that users be able to upgrade the version of `dvipng` that is linked to your app. Since iOS doesn’t allow .dylib files, and Code Signing wouldn’t allow a user to run a modified binary anyway, you will have to make the source of your app available to all of your users. And that’s assuming the LGPL doesn’t have the same restriction listed in the link above preventing you from releasing the app (via the App Store) in the first place.

“I’m not a lawyer” and all that, but that’s my understanding.

**Comment by sincebelieve** [comment]
Thanks for the information, I'm a newbie in this industry, I think I need a more serious survey on software licences.

I successfully port TeX– to Android, I think people may be interested, here is the configuration for cross-compiling (on Mac OSX 10.9):

In Android.mk:

``` makefile
LOCAL_PATH := $(call my-dir)

include $(CLEAR_VARS)

LOCAL_MODULE    := tex-engine      # => libtex-engine.so
LOCAL_SRC_FILES := tex-engine.cpp
LOCAL_CPPFLAGS  += -std=c++11

include $(BUILD_SHARED_LIBRARY)
```

In Application.mk:

``` makefile
NDK_TOOLCHAIN_VERSION := clang
APP_STL := gnustl_static
```
